test/ffmpeg-vaapi/encode/encoder.py

```diff
@@ -107,19 +117,31 @@ def encode(self):
 
     vars(self).update(rcmodeu = self.rcmode.upper())
 
-    self.encoded = get_media()._test_artifact(
-      "{}.{}".format(self.gen_name(), self.get_file_ext()))
-
-    iopts = self.gen_input_opts()
-    oopts = self.gen_output_opts()
-
-    self.output = call(
-      "ffmpeg -hwaccel vaapi -vaapi_device /dev/dri/renderD128 -v verbose"
-      " {iopts} {oopts}".format(iopts = iopts, oopts = oopts))
-
-    self.check_output()
-    self.check_bitrate()
-    self.check_metrics()
+    if vars(self).get("r2r", None) is not None:
+      assert (self.r2r > 1) is True, "invalid r2r value"
+      r2r_md5 = []
+      vars(self).setdefault("metric", dict(type = "md5"))
+
+      for self.count in xrange(self.r2r):
+        self.encoded = get_media()._test_artifact(
+        "{}_{}.{}".format(self.gen_name(), self.count, self.get_file_ext()))
+
+        self.gen_output()
+        md5 = self.gen_r2r_md5()
+        slash.logger.info("md5 for r2r {} is {}".format(self.count, md5))
```

Comment:
Use `get_media()._set_test_details` instead.  See example in other comment.

Reply:
modified to use get_media()._set_test_details